library/ssl_msg.c

```diff
@@ -5635,7 +5632,7 @@ void mbedtls_ssl_read_version( int *major, int *minor, int transport,
         *major = 255 - ver[0] + 2;
         *minor = 255 - ver[1] + 1;
 
-        if( *minor == MBEDTLS_SSL_MINOR_VERSION_1 )
+        if( *minor == MBEDTLS_SSL_MINOR_VERSION_3 - 2 )
```

Comment:
deadcode, remove this `if` and it's code block

Reply:
This is not a dead code! The reason for this part is that the TLS client has to recognize the older version on the other side to be able to convey it to the calling function `ssl_parse_hello_verify_request()` in `ssl_cli.c` to allow it to properly react as it's (not clearly) specified in the RFC - see my comment below on `library/ssl_cli.c`:1881. Please tell if you need me to resolve it in another way.